azure-mgmt-servicebus/src/main/java/com/microsoft/azure/management/servicebus/Namespace.java

```diff
@@ -0,0 +1,280 @@
+     */
+    NamespaceAuthorizationRules authorizationRules();
+
+    /**
+     * The entirety of the service bus namespace definition.
+     */
+    interface Definition extends
+            Namespace.DefinitionStages.Blank,
+            Namespace.DefinitionStages.WithGroup,
+            Namespace.DefinitionStages.WithCreate {
+    }
+
+    /**
+     * Grouping of service bus namespace definition stages.
+     */
+    interface DefinitionStages {
+        /**
+         * The first stage of a service bus namespace definition.
+         */
+        interface Blank extends GroupableResource.DefinitionWithRegion<WithGroup> {
+        }
+
+        /**
+         * The stage of the service bus namespace definition allowing to specify the resource group.
+         */
+        interface WithGroup extends GroupableResource.DefinitionStages.WithGroup<WithCreate> {
+        }
+
+        /**
+         * The stage of the service bus namespace definition allowing to specify the sku.
+         */
+        interface WithSku {
+            /**
+             * Specifies the namespace sku.
+             *
+             * @param namespaceSku the sku
+             * @return next stage of the service bus namespace definition
+             */
+            WithCreate withSku(NamespaceSku namespaceSku);
+        }
+
+        /**
+         * The stage of the service bus namespace definition allowing to add a new queue in the namespace.
+         */
+        interface WithQueue {
+            /**
+             * Creates a queue entity in the service bus namespace.
+             *
+             * @param name queue name
+             * @param maxSizeInMB maximum size of memory allocated for the queue entity
+             * @return next stage of the service bus namespace definition
+             */
+            WithCreate withNewQueue(String name, int maxSizeInMB);
+        }
+
+        /**
+         * The stage of the service bus namespace definition allowing to add a new topic in the namespace.
+         */
+        interface WithTopic {
+            /**
+             * Creates a topic entity in the service bus namespace.
+             *
+             * @param name topic name
+             * @param maxSizeInMB maximum size of memory allocated for the topic entity
+             * @return next stage of the service bus namespace definition
+             */
+            WithCreate withNewTopic(String name, int maxSizeInMB);
+        }
+
+        /**
+         * The stage of the service bus namespace definition allowing to add an authorization rule for accessing
+         * the namespace.
+         */
+        interface WithAuthorizationRule {
+            /**
+             * Creates a send authorization rule for the service bus namespace.
+             *
+             * @param name rule name
+             * @return next stage of the service bus namespace definition
+             */
+            WithCreate withNewSendRule(String name);
```

Comment:
What we do here is - we create a rule CHILD resource (external child resource) that enables sending message. Later user can retrieve this rule using `namespace.authorizationRules()` and retrieve `connection string/key` from the rule using which message can be send. I think we should somehow make it clear that we are creating a rule resource with the given NAME. `withSendingAllowed()` sounds more like we are turning a flag on in the `namespace` resource. What do you think?